wallet-new/test/unit/Test/Spec/Wallets.hs

```diff
@@ -311,8 +313,9 @@ spec = describe "Wallets" $ do
         describe "Wallet update password (kernel)" $ do
             prop "correctly replaces the ESK in the keystore" $ withMaxSuccess 50 $
                 monadicIO $ do
-                    newPwd <- pick arbitrary
+                    newPwd <- pick $ arbitrary `suchThat` (/= emptyPassphrase)
```

Comment:
I would suggest not to touch the logic of this test, as this properly tests only one thing, which is the keystore replacement. What am I missing? What's the rationale behind the change?

Reply:
I basically duplicated this test, checked for `HasSpendingPassword` here, and `NoSpendingPassword` in the new one.

I considered re-calculating the `HasSpendingPassword` from the arbitrary password [here](https://github.com/input-output-hk/cardano-sl/pull/3621/commits/df5a3ddd077ced9c3b9a5b38299d58385c36b652#diff-c76387a32496e0a7d02ba0d70564f7d0R336) making sure it was the same as the one in the hdRoot, but that felt to implicit, and dependent on `arbitrary` picking both empty and non-empty passwords.

I could make another copy? (Also, should I strip away the checks already tested here?)

Reply:
My (relatively small) concern is that the first property is called "correctly replaces the ESK in the keystore" and, as such, it should work regardless of which spending password we pick to begin with. Right now it feels like this first test is a bit misleading, this is why I have proposed to leave it be and simply add another one (like you did) to check for this specific concern only.